GeeksCoreLibrary/Modules/PostalServices/PostNL/Services/PostNLService.cs

```diff
@@ -0,0 +1,194 @@
+﻿using GeeksCoreLibrary.Modules.PostalServices.PostNL;
+using GeeksCoreLibrary.Modules.Databases.Interfaces;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using RestSharp;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using GeeksCoreLibrary.Core.DependencyInjection.Interfaces;
+using GeeksCoreLibrary.Core.Models;
+using GeeksCoreLibrary.Modules.Objects.Interfaces;
+using GeeksCoreLibrary.Modules.PostalServices.PostNL.Interfaces;
+using GeeksCoreLibrary.Modules.PostalServices.PostNL.Models;
+
+namespace GeeksCoreLibrary.Modules.PostalServices.PostNL.Services
+{
+    public class PostNLService : IPostNLService, IScopedService
+    {
+        private readonly IDatabaseConnection databaseConnection;
+        private readonly GclSettings gclSettings;
+        private readonly IObjectsService objectService;
+
+        public PostNLService(ILogger<PostNLService> logger,
+            IDatabaseConnection databaseConnection,
+            IOptions<GclSettings> gclSettings,
+            IObjectsService objectService)
+        {
+            this.databaseConnection = databaseConnection;
+            this.gclSettings = gclSettings.Value;
+            this.objectService = objectService;
+        }
+
+        public static List<string> EuropeanCountries = new() { "AT", "IT", "BE", "LV", "BG", "LT", "HR", "LU", "CY", "CZ", "DK", "EE", "PL", "FI", "PT", "FR", "RO", "DE", "SK", "SI", "GR", "ES", "HU", "SE", "IE" };
+
+        public enum ShippingLocations
+        {
+            Netherlands,
+            Europe,
+            Global
+        }
+
+        /// <summary>
+        /// Cleans the PostNL log table
+        /// </summary>
+        private async Task CleanLogs()
+        {
+            await databaseConnection.ExecuteAsync("DELETE FROM cust_postnl_log WHERE datetime < DATE_SUB(NOW(), INTERVAL 2 WEEK)");
```

Comment:
In de GCL willen we geen tabelnamen direct in query's hebben staan. Kun je hier een constante van maken en dan die constante in de query zetten?

Reply:
Daarnaast ook graag deze tabel een andere naam geven. De prefix `cust_` is, zoals de naam al zegt, voor custom tabellen bij klanten. Tabellen die de GCL/JCL gebruikt mogen deze prefix nooit hebben. Deze tabel kan dan gewoon `postnl_log` heten bijvoorbeeld.